components/supervisor/pkg/supervisor/supervisor.go

```diff
@@ -1006,8 +1010,9 @@ func isBlacklistedEnvvar(name string) bool {
 	return false
 }
 
-func startAPIEndpoint(ctx context.Context, cfg *Config, wg *sync.WaitGroup, services []RegisterableService, tunneled *ports.TunneledPortsService, opts ...grpc.ServerOption) {
+func startAPIEndpoint(ctx context.Context, cfg *Config, wg *sync.WaitGroup, readyCallback context.CancelFunc, services []RegisterableService, tunneled *ports.TunneledPortsService, opts ...grpc.ServerOption) {
```

Comment:
I don't understand the purpose/use of the "readyCallback". Also, why is this a `CancelFunc`?